plugins/Files/js/browser.js

```diff
@@ -226,34 +303,116 @@ var browser = {
 		// Delete files and file elements
 		files.forEach(function(file) {
 			file.delete(function() {
-				$('#' + file.name).remove();
+				$('#' + file.hashedPath).remove();
 			});
 		});
 	},
 
+	// Prompts user for Ascii or .sia method of sharing
+	shareSelected () {
+		var files = getSelectedFiles();
+		var itemCount = files.length;
+		var label;
+		var paths;
+
+		// Check for any selected files, and make messages singular or plural
+		// Also get paths of selected files, including files several layers
+		// inside a folder, in a one dimensional array
+		if (itemCount === 0) {
+			tools.tooltip('No selected files', $('.controls .share').get(0));
+			return;
+		} else if (itemCount === 1) {
+			let file = files[0];
+			label = file.name;
+			paths = file.type === 'folder' ? file.paths : [file.path];
+		} else {
+			// Reduce array of files to array of all file paths
+			paths = files.reduce(function(a, b) {
```

Comment:
reducing an array to an array sounds like a map, no? Or are the types different?